Domain/Utilities/AdminHandler.cs

```diff
@@ -0,0 +1,165 @@
+                        ReadKey();
+                        break;
+                    case "2":
+                        ImportFlightsFromCsv();
+                        ReadKey();
+                        break;
+                    case "3":
+                        exit = true;
+                        break;
+                    default:
+                        Console.WriteLine("Invalid option. Please try again.");
+                        break;
+                }
+            }
+        }
+
+        private bool AuthenticateUser()
+        {
+            string correctUsername = "admin";
+            string correctPassword = "1234";
+
+            Console.WriteLine("Please enter your username:");
+            string username = Console.ReadLine();
+
+            Console.WriteLine("Please enter your password:");
+            string password = Console.ReadLine();
+
+            if (username == correctUsername && password == correctPassword)
+            {
+                Console.WriteLine("Authentication successful!");
+                return true;
+            }
+            else
+            {
+                Console.WriteLine("Invalid username or password. Please try again.");
+                return false;
+            }
+        }
+
+        private void SearchAndDisplayBookings()
+        {
+            Console.Write("Enter Departure Country: ");
+            string departureCountry = Console.ReadLine()?.Trim();
+
+            Console.Write("Enter Destination Country: ");
+            string destinationCountry = Console.ReadLine()?.Trim();
+
+            Console.Write("Enter Departure Airport: ");
+            string departureAirport = Console.ReadLine()?.Trim();
+
+            Console.Write("Enter Arrival Airport: ");
+            string arrivalAirport = Console.ReadLine()?.Trim();
+
+            Console.Write("Enter Departure Date (yyyy-MM-dd) or leave empty: ");
+            string dateInput = Console.ReadLine()?.Trim();
+            DateTime? departureDate = string.IsNullOrEmpty(dateInput) ? null : DateTime.Parse(dateInput);
+
+            Console.Write("Enter Passenger Name or leave empty: ");
+            string passenger = Console.ReadLine()?.Trim();
+
+            string seatClass = InputHelper.ValidateSeatClass();
+
+            Console.Write("Enter Maximum Price or leave empty: ");
+            string priceInput = Console.ReadLine()?.Trim();
+            double? maxPrice = string.IsNullOrEmpty(priceInput) ? null : double.Parse(priceInput);
+
+            var bookings = _bookingService.FilterBookings(maxPrice, departureCountry, destinationCountry, departureDate, departureAirport, arrivalAirport, passenger, seatClass);
+
+            if (bookings.Any())
+            {
+                Console.WriteLine("\nFiltered Bookings based on your search:");
+                foreach (var booking in bookings)
+                {
+                    var flight = _flightService.GetFlightById(booking.FlightId);
```

Comment:
retrieving flights info in a for loop would be expensive. On line 123, `FilterBookings` already has logic that retrieves the bookings and their flights, maybe you can think about returning the flight info along side the booking returned.